crates/radix-engine-toolkit/src/transaction_types/interface.rs

```diff
@@ -300,6 +302,29 @@ pub fn execution_summary(
         royalty_cost: receipt.fee_summary.total_royalty_cost_in_xrd,
     };
 
+    for (node_id, states_to_update) in
+        receipt.expect_commit_ignore_outcome().state_updates()
+    {
+        if node_id.is_global_non_fungible_resource_manager() {
```

Comment:
I think that a disadvantage of this approach is that we can't distinguish between newly minted NFTs and updates to the non-fungible data. It might be better to get this information from events due to that. Also, this might be a good test case to add, non-fungible data updates are not picked up as newly minted non-fungibles.

Reply:
I've implemented logic based on events.

Reply:
Also I've implemented `radix-engine-toolkit-uniffi` part.